Disable read/search db replicas in TE/E0

jespino opened the pull request:
#### Summary
Restrict the usage of read/search replicas to E10 and E20

#### Ticket Link
[MM-24618](https://mattermost.atlassian.net/browse/MM-24618)

mattermod commented:
Triggered enterprise tests.

jespino commented on config/utils.go:
```diff
@@ -35,22 +35,27 @@ func desanitize(actual, target *model.Config) {
-	if *target.SqlSettings.AtRestEncryptKey == model.FAKE_SETTING {
-		target.SqlSettings.AtRestEncryptKey = actual.SqlSettings.AtRestEncryptKey
+
+	if len(target.SqlSettings.DataSourceReplicas) == 1 && target.SqlSettings.DataSourceReplicas[0] == model.FAKE_SETTING {
```
This is because before it was overwriting always this, so you can't change this value from the outside. Now it is de-sanitized only if was a FAKE_SETTING, it doesn't impact the interface (for now) because is not shown, but from the API perspective, suddenly you are receiving a `[*************************************]` in the Read replicas and Search replicas. I guess is not a big deal.

coreyhulen commented on app/server.go:
```diff
@@ -275,10 +275,17 @@ func NewServer(options ...Option) (*Server, error) {
+		mlog.Warn("Read replicas functionality disabled by current license. Please contact your system administrator about upgrading your enterprise license.")
 		s.UpdateConfig(func(cfg *model.Config) {
-			cfg.SqlSettings.DataSourceReplicas = cfg.SqlSettings.DataSourceReplicas[:1]
+			cfg.SqlSettings.DataSourceReplicas = []string{}
```
Should this be set to blank or to master?

jespino replied:
It is perfectly find with this, because the store check for replicas and if the length of replicas is 0 use master when you ask for `GetReplica`

coreyhulen commented:
Any documentation that needs to change?

jespino commented:
@coreyhulen actually the documentation is always talking about read replicas and search replicas as an enterprise feature: https://docs.mattermost.com/deployment/deployment.html#multiple-read-replicas-enterprise-edition, https://docs.mattermost.com/administration/config-settings.html#read-replicas, and two other mentions as part of the clustering/high availability documentation, so clearly related to E20.

mattermod commented:
Triggered enterprise tests.

catalintomai commented on app/server_test.go:
```diff
@@ -36,6 +37,62 @@ func TestStartServerSuccess(t *testing.T) {
+		require.Len(t, s.Config().SqlSettings.DataSourceSearchReplicas, 0)
+	})
+
+	t.Run("Search Read Replicas With License", func(t *testing.T) {
```
remove "Read"?

jespino replied:
makes sense

lindy65 commented:
Hi @jespino :)

Please help add test steps here or on the Jira ticket?  Thanks!

jespino commented:
@lindy65 Added, maybe this QA testing is for somebody with more technical experience because you have to setup a read replica.

lindy65 commented:
Thanks @jespino - I'll ask one of the QA SDETs to take over review of this PR and switch out the reviewer once I know who it'll be 👍

jespino commented:
@srkgupta thanks, yes, you are right, that code was already there, and wasn't doing the job correctly before. Now I moved it before the database initialization, and actually I added some tests to verify it. But now should be working.

jespino commented:
@srkgupta I going to check it right away

jespino commented:
@srkgupta Ok, I think now should be working, the license was loaded later in the process, so now, what I do is disconnect and reconnect again if is needed because the read replicas are removed.

justinegeffen commented:
@jespino, confirming that docs are currently aligned with this change and no updates are required? Docs are: https://docs.mattermost.com/deployment/deployment.html#multiple-read-replicas-enterprise-edition and https://docs.mattermost.com/administration/config-settings.html#read-replicas. 

Thanks! :)

jespino commented:
@justinegeffen yes, it was already that way, this PR only make the system consistent with was previously documented.

justinegeffen commented:
Awesome, thank you! :)
